script.sh

```diff
@@ -1,49 +1,105 @@
 #!/bin/sh
 set -e
 
-NOW_TOKEN_OPTION="--token=$PLUGIN_TOKEN"
-NOW_DEPLOY_OPTIONS="${NOW_TOKEN_OPTION} --no-clipboard"
+NOW_DEPLOY_OPTIONS=" --no-clipboard"
 
-if [ -n "$PLUGIN_NAME" ]
+# Get the token or error
+if [ -z "$PLUGIN_NOW_TOKEN" ]
 then
-    NOW_DEPLOY_OPTIONS="${NOW_DEPLOY_OPTIONS} --name ${PLUGIN_NAME}"
+    # No explicit token provided, check for secret
+    if [ -z "$NOW_TOKEN" ]
+    then
+        echo "> Error!! either the parameter now_token or the secret NOW_TOKEN is required!"
+        exit 1;
+    else
+        PLUGIN_NOW_TOKEN="$NOW_TOKEN"
+    fi
+fi
+
+# Get the team if provided
+if [ -n "$PLUGIN_TEAM" ]
+then
+    echo "> adding team $PLUGIN_TEAM"
+    NOW_TEAM_OPTION="--team $PLUGIN_TEAM"
+else
+    echo "> No team name provided."
+fi
+
+NOW_AUTH="$NOW_AUTH --token $PLUGIN_NOW_TOKEN $NOW_TEAM_OPTION"
+
+if [ -n "$PLUGIN_DEPLOY_NAME" ]
+then
+    echo "> adding deploy_name $PLUGIN_DEPLOY_NAME"
+    NOW_DEPLOY_OPTIONS="${NOW_DEPLOY_OPTIONS} --name $PLUGIN_DEPLOY_NAME"
+else
+    echo "> No deployment name provided. The directory will be used as the name"
 fi
 
 if [ -n "$PLUGIN_TYPE" ]
 then
-    NOW_DEPLOY_OPTIONS="${NOW_DEPLOY_OPTIONS} --${PLUGIN_TYPE}"
+    echo "> adding type $PLUGIN_TYPE"
+    NOW_DEPLOY_OPTIONS="${NOW_DEPLOY_OPTIONS} --$PLUGIN_TYPE"
+else
+    echo "> No deployment type provided, now.sh will try to detect it..."
 fi
 
-if [ -n "$PLUGIN_TOKEN" ]
+if [ -n "$NOW_AUTH" ]
 then
-    echo "> Deploying on now.sh…" &&
-    NOW_DEPLOYMENT_URL=$(now $NOW_DEPLOY_OPTIONS) &&
-    echo "> Success! Deployment complete to $NOW_DEPLOYMENT_URL!";
+    DEPLOY_DIR="$PLUGIN_DIRECTORY"
+    # Let the user know which directory they are deploying
+    if [ -z "$DEPLOY_DIR" ]
+    then
+        DEPLOY_DIR="."
+    fi
+    NOW_DEPLOY_OPTIONS="${NOW_DEPLOY_OPTIONS}"
+    echo "> Deploying $DEPLOY_DIR on now.sh…" &&
+    NOW_DEPLOYMENT_URL=$(now $NOW_AUTH $NOW_DEPLOY_OPTIONS $PLUGIN_DIRECTORY) &&
```

Comment:
Here the deploy command is not surrounded by a conditional as it was before. I can't think of a case when $NOW_AUTH wouldn't already exist, but it might be safer to do a second check for  `if [ -n "$NOW_AUTH" ]` here and throw a descriptive log message? If you have more ideas on this feel free to request more changes